python-backend/app/api/mines/compliance/models/compliance_article.py

```diff
@@ -0,0 +1,48 @@
+from datetime import datetime
+from app.extensions import db
+
+from ....utils.models_mixins import AuditMixin, Base
+
+
+class ComplianceArticle(AuditMixin, Base):
+    __tablename__ = 'compliance_article'
+    compliance_article_id = db.Column(db.Integer, nullable=False, primary_key=True)
+    article_act_code = db.Column(db.String(5), nullable=False)
+    section = db.Column(db.String(6), nullable=True)
+    sub_section = db.Column(db.String(4), nullable=True)
+    paragraph = db.Column(db.String(4), nullable=True)
+    sub_paragraph = db.Column(db.String(4), nullable=True)
+    description = db.Column(db.String(100), nullable=False)
+    effective_date = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+    expiry_date = db.Column(db.DateTime, nullable=False, default=datetime.strptime('9999-12-31', '%Y-%m-%d'))
+
+    def __repr__(self):
+        return '<ComplianceArticle %r>' % self.compliance_article_id
+
+
+    @classmethod
+    def all_options(cls):
```

Comment:
Is there a reason that this doesn't follow our current pattern for models and returns from models?

Reply:
You'll have to be a bit more explicit about the pattern you're referring to. This is the exact pattern followed by the MineCommodityCode model, which (like this model) is for a code table. This pattern is also used in the SubDivisionCode model, MineDisturbanceCode model, and MineTenureTypeCode model.

Perhaps you're thinking of models that don't reflect code tables?

Reply:
Should this table have the "code" suffix then?

Reply:
I think there are two patterns going for status code tables at the moment. Application_status_codes, and Permit_status_codes are returned in the standard way all of our other models return.

Reply:
James, I wondered about that. This was the table name that I was given in discussions with Kristin, so I implemented it. Totally willing to reassess the name

Justin, I can take a look at those. I don't think we can say one is more 'standard' than the other at this point considering the number of other code tables implemented in this way

Reply:
I agree, I just think we need to decide on one going forward to keep everything consistent.